### ISSUE TITLE
[Bug]

### ISSUE THREAD

**Issue by emjayess** [webinstall/webi-installers#782]
## What were you trying to install (or what else went wrong)?

webi itself (first-time webi user)

## What exactly did you do?

I ran `curl -sS https://webi.sh/ | sh` to install webi, per the site

## What went wrong?

It completed the installation successfully, and `webi` is available. However the installation output, as well the output of subsequent `webi` invocations, is littered with a warning that `uname -o` is an illegal option, which seems to be in the `my_libc` detection block of the script...

```
uname: illegal option -- o
usage: uname [-amnprsv]
```

Seen here 3x, in the installation output:
```zsh
>>> Welcome to Webi! - modern tools, instant installs.  <<<
    We expect your experience to be absolutely perfect!

    Success? Star it!   https://github.com/webinstall/webi-installers
    Problem? Report it: https://github.com/webinstall/webi-installers/issues
uname: illegal option -- o
usage: uname [-amnprsv]
                        (your system is Darwin/arm64 with libc & curl+wget)

Bootstrapping Webi
    Downloading https://webi.sh/packages/webi/webi.sh
        to ~/.local/bin/webi
uname: illegal option -- o
usage: uname [-amnprsv]
    Running ~/.local/bin/webi webi@stable

uname: illegal option -- o
usage: uname [-amnprsv]
Installing webi ...
    Found  ~/.local/bin
    Initializing ~/.config/envman/
    Edit ~/.zshrc to source ~/.config/envman/load.sh

'webi' installed to ~/.local/bin/webi

Usage:
    webi <package-name>[@version] ...

Example:
    webi node@lts prettier vim-essentials

    Edit ~/.config/envman/PATH.env to add:
        ~/.local/bin

>>> ACTION REQUIRED <<<
        Copy, paste & run the following command:
        source ~/.config/envman/PATH.env
        (newly opened terminal windows will update automatically)
```

And also seen here in output of e.g. `webi --version`:
```zsh
uname: illegal option -- o
usage: uname [-amnprsv]
webi v1.2.8 (1ab2245d) Copyright 2020+ AJ ONeal
    https://webinstall.dev/webi
```

## Which OS did you try on?

- [x] Mac (m1 macbook air)
- [ ] Windows
- [ ] Linux

## What type of computer (i.e. laptop, desktop, Raspberry Pi)?

- [x] Laptop or Desktop (amd64 it's a M1 macbook air – macOS Monterey 12.7.1)

`uname -amnprsv`:
```
Darwin macaronic-air.home 21.6.0 Darwin Kernel Version 21.6.0: Wed Oct  4 23:54:48 PDT 2023; root:xnu-8020.240.18.704.15~1/RELEASE_ARM64_T8101 arm64 arm
```

**Comment by coolaj86** [comment]
That's very interesting. Newer versions of macOS do support the -o option.

That said, we've had this problem on some of the other BSDs and it will be fixed as part of #756.

**Comment by coolaj86** [comment]
Thanks for the report.

I'll close this since the fix is already in an active PR.
